File: Components/RepositoryAttributes/RepositoryAttributesInfo.vb
Imports System
Imports System.Configuration
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports System.Reflection
Imports System.Xml
Imports DotNetNuke
Imports DotNetNuke.Common.Utilities
Imports DotNetNuke.Common.Globals
Imports DotNetNuke.Services.Search

Namespace DotNetNuke.Modules.Repository
    Public Class RepositoryAttributesInfo

        Public Enum ChangeStates
            UNCHANGED = 0
            ADD = 1
            EDIT = 2
            DELETE = 3
        End Enum

        Private _itemID As Integer
        Private _moduleID As Integer
        Private _attributeName As String
        Private _changeState As ChangeStates = ChangeStates.UNCHANGED

        Public Sub New()
        End Sub

        Public Sub New(ByVal moduleID As Integer, ByVal attributeName As String)
            _moduleID = moduleID
            _attributeName = attributeName
        End Sub

        Public Property ItemID() As Integer
            Get
                Return _itemID
            End Get
            Set(ByVal Value As Integer)
                _itemID = Value
            End Set
        End Property

        Public Property ModuleID() As Integer
            Get
                Return _moduleID
            End Get
            Set(ByVal Value As Integer)
                _moduleID = Value
            End Set
        End Property

        Public Property AttributeName() As String
            Get
                Return _attributeName
            End Get
            Set(ByVal Value As String)
                _attributeName = Value
            End Set
        End Property

        Public Property ChangeState() As ChangeStates
            Get
                Return _changeState
            End Get
            Set(ByVal Value As ChangeStates)
                _changeState = Value
            End Set
        End Property

    End Class
End Namespace

File: Components/RepositoryAttributes/RepositoryAttributesInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Configuration;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using System.Reflection;
using System.Xml;
using DotNetNuke;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Common;
using DotNetNuke.Services.Search;

namespace DotNetNuke.Modules.Repository
{
	public class RepositoryAttributesInfo
	{

		public enum ChangeStates
		{
			UNCHANGED = 0,
			ADD = 1,
			EDIT = 2,
			DELETE = 3
		}

		private int _itemID;
		private int _moduleID;
		private string _attributeName;

		private ChangeStates _changeState = ChangeStates.UNCHANGED;
		public RepositoryAttributesInfo()
		{
		}

		public RepositoryAttributesInfo(int moduleID, string attributeName)
		{
			_moduleID = moduleID;
			_attributeName = attributeName;
		}

		public int ItemID {
			get { return _itemID; }
			set { _itemID = value; }
		}

		public int ModuleID {
			get { return _moduleID; }
			set { _moduleID = value; }
		}

		public string AttributeName {
			get { return _attributeName; }
			set { _attributeName = value; }
		}

		public ChangeStates ChangeState {
			get { return _changeState; }
			set { _changeState = value; }
		}

	}
}
